Quaver.Shared/Screens/Edit/UI/Panels/EditorPanelColorPicker.cs

```diff
@@ -0,0 +1,189 @@
+                Position = new ScalableVector2(Padding, Padding)
+            };
+
+            CreateSnapItems(LeftList, new[] { SnapColor.Red, SnapColor.Blue, SnapColor.Yellow, SnapColor.Orange, SnapColor.Green });
+        }
+
+        private void CreateRightList()
+        {
+            RightList = new Container()
+            {
+                Parent = Content,
+                Alignment = Alignment.TopRight,
+                Size = new ScalableVector2(Content.Width / 2 - 2 * Padding, Content.Height - 2 * Padding),
+                Position = new ScalableVector2(-Padding, Padding)
+            };
+
+            CreateSnapItems(RightList, new[] { SnapColor.Purple, SnapColor.Pink, SnapColor.Cyan, SnapColor.White, SnapColor.None });
+        }
+
+        private void CreateSnapItems(Container parent, SnapColor[] snaps)
+        {
+            for (var i = 0; i < snaps.Length; i++)
+            {
+                new DrawableEditorColorPicker(snaps[i], SelectedHitObjects, ActionManager)
+                {
+                    Parent = parent,
+                    Size = new ScalableVector2(parent.Width, (parent.Height / snaps.Length) - 2.5f),
+                    Y = i * parent.Height / snaps.Length,
+                };
+            }
+        }
+    }
+
+    public class DrawableEditorColorPicker : ImageButton
+    {
+        /// <summary>
+        /// </summary>
+        public SnapColor Color { get; }
+
+        private BindableList<HitObjectInfo> SelectedHitObjects { get; }
+        private EditorActionManager ActionManager { get; }
+        private Sprite Line { get; set; }
+        private SpriteTextPlus Name { get; set; }
+        private SpriteTextPlus SnapText { get; set; }
+
+        /// <inheritdoc />
+        /// <summary>
+        /// </summary>
+        /// <param name="colorIndex"></param>
+        /// <param name="selectedHitObjects"></param>
+        /// <param name="manager"></param>
+        public DrawableEditorColorPicker(SnapColor colorIndex, BindableList<HitObjectInfo> selectedHitObjects,
+            EditorActionManager manager) : base(UserInterface.OptionsSidebarButtonBackground)
+        {
+            Color = colorIndex;
+            SelectedHitObjects = selectedHitObjects;
+            ActionManager = manager;
+
+            CreateLine();
+            CreateName();
+
+            Alpha = 0;
+            Clicked += OnClicked;
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void OnClicked(object sender, EventArgs e)
+        {
+            if (SelectedHitObjects.Value.Count == 0)
+            {
+                NotificationManager.Show(NotificationLevel.Warning, "You need to select objects before changing the color!");
+                return;
+            }
+
+            ActionManager.Perform(new EditorActionSetColor(ActionManager, new List<HitObjectInfo>(SelectedHitObjects.Value), (int)Color));
```

Comment:
Since we have a SnapColor enum in Quaver.API, we might as well use it here and anywhere else that deals with snap colors